videostore/concat.c

```diff
@@ -129,18 +129,27 @@ int video_store_concat(const char *concat_filepath, const char *output_path) {
         AV_ROUND_NEAR_INF | AV_ROUND_PASS_MINMAX);
     packet->pos = -1;
 
+    if (packet->dts <= prevDts) {
```

Comment:
It turns out that when the concatenator goes from processing the last packet of one file to processing the first packet of the next file the dts will be repeated and `AV_PKT_FLAG_DISCARD` won't be set on the pkt->flags. I think `AV_PKT_FLAG_DISCARD` is only set if you are concatenating a subset of a file. From what I can tell if you are concatenating entire files, it won't be set.

Previously in my testing I was only testing 30 second ranges and as a result there were only either 1 whole file or 2 partial files being concatenated, one of which had `AV_PKT_FLAG_DISCARD`  set.

Now we need to explicitly check for repeating dts and skip it.

Reply:
I also confirmed this doensn't result in any hitching or artifacts in the videos produced, both in h264 and h265

Reply:
1. They have been equivalent in all the cases I've seen but I saw in the gortsplib examples of writing an ts file to disk they pass dts and pts separately, so I followed that pattern here in case there are cameras out there which will send dts that is different than pts.

https://github.com/bluenviron/gortsplib/blob/main/examples/client-play-format-h265-to-disk/mpegts_muxer.go#L105